doc/api-hidden.rst

```diff
@@ -27,13 +27,66 @@
    Dataset.std
    Dataset.var
 
+   core.rolling.DatasetCoarsen
```

Comment:
Can you add this to `api.rst` with a top-level heading "Coarsen objects" like "GroupBy objects" etc.